src/monitor/node_metadata.c

```diff
@@ -276,6 +286,56 @@ GetAutoFailoverNode(char *nodeName, int nodePort)
 }
 
 
+/*
+ * GetAutoFailoverNodeWithId returns a single AutoFailover
+ * identified by nodeid.
+ */
+AutoFailoverNode *
+GetAutoFailoverNodeWithId(int nodeid)
+{
+	AutoFailoverNode *pgAutoFailoverNode = NULL;
+	MemoryContext callerContext = CurrentMemoryContext;
+
+	Oid argTypes[] = {
+		INT4OID  /* nodeport */
+	};
+
+	Datum argValues[] = {
+		Int32GetDatum(nodeid)        /* nodeid */
+	};
+	const int argCount = sizeof(argValues) / sizeof(argValues[0]);
+	int spiStatus = 0;
+
+	const char *selectQuery =
+		SELECT_ALL_FROM_AUTO_FAILOVER_NODE_TABLE
+		" WHERE nodeid = $1";
```

Comment:
Do we need to add nodename and nodeport to the WHERE clause here? I think I remember that's what we do elsewhere, to prevent things from having changed concurrently.

Reply:
do you suggest filtering based on (nodeId, nodename, nodeport) ?
I originally had the latter two here, then you asked me to do checks based on nodeid.  Can  you clarify ?